distribution/tools/upgrade-cli/src/main/java/org/opensearch/upgrade/InstallPluginsTask.java

```diff
@@ -0,0 +1,118 @@
+ * compatible open source license.
+ */
+
+package org.opensearch.upgrade;
+
+import org.opensearch.cli.Terminal;
+import org.opensearch.common.collect.Tuple;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Installs the list of plugins using the opensearch-plugin command.
+*/
+class InstallPluginsTask implements UpgradeTask {
+    private static final String ERROR_MSG = "Error installing plugin %s. Please install it manually.";
+
+    /** The list of official plugins that can be installed by the upgrade tool. */
+    static final Set<String> OFFICIAL_PLUGINS;
+    static {
+        try (
+            InputStream stream = InstallPluginsTask.class.getResourceAsStream("/plugins.txt");
+            BufferedReader reader = new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8))
+        ) {
+            Set<String> plugins = new HashSet<>();
+            String line = reader.readLine();
+            while (line != null) {
+                plugins.add(line.trim());
+                line = reader.readLine();
+            }
+            OFFICIAL_PLUGINS = Collections.unmodifiableSet(plugins);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void accept(final Tuple<TaskInput, Terminal> input) {
+        final TaskInput taskInput = input.v1();
+        final Terminal terminal = input.v2();
+        if (taskInput.getPlugins() == null || taskInput.getPlugins().isEmpty()) {
+            return;
+        }
+        terminal.println("Installing core plugins ...");
+        List<String> manualPlugins = new ArrayList<>();
+
+        for (String plugin : taskInput.getPlugins()) {
+            if (OFFICIAL_PLUGINS.contains(plugin)) {
+                executeInstallPluginCommand(plugin, taskInput, terminal);
+            } else {
+                manualPlugins.add(plugin);
+            }
+        }
+        if (!manualPlugins.isEmpty()) {
+            terminal.println("Please install the following custom plugins manually: " + manualPlugins);
+        }
+        terminal.println("Success!" + System.lineSeparator());
+    }
+
+    // package private for unit testing
+    void executeInstallPluginCommand(String plugin, TaskInput taskInput, Terminal terminal) {
+        ProcessBuilder processBuilder = getProcessBuilderBasedOnOS(plugin, taskInput);
+        try {
+            final Process process = processBuilder.inheritIO().start();
+            if (process.waitFor() != 0) {
+                terminal.errorPrint(Terminal.Verbosity.NORMAL, String.format(Locale.getDefault(), ERROR_MSG, plugin));
+            }
+        } catch (IOException | InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)